patches/net/minecraft/world/entity/LivingEntity.java.patch

```diff
@@ -48,6 +56,12 @@
      }
  
      @Override
+     protected void checkFallDamage(double p_20990_, boolean p_20991_, BlockState p_20992_, BlockPos p_20993_) {
+-        if (!this.isInWater()) {
++        if (!this.isInWater() && !this.isInFluid()) { // Neo: only do fluid pushing if it wasn't previously calculated
+             this.updateInWaterStateAndDoWaterCurrentPushing();
```

Comment:
Copying from discord:

I think the better solution is to patch updateInWaterStateAndDoWaterCurrentPushing to take a boolean to control whether to run the fluid pushing logic or not. Something like this. Especially as all this code really needs to do is update the fluid states, not run pushing
<img width="1330" height="760" alt="image" src="https://github.com/user-attachments/assets/f3d2499f-6526-40a0-bba9-8422e4411add" />